vizro-core/src/vizro/models/_components/ag_grid.py

```diff
@@ -44,10 +44,17 @@ class AgGrid(VizroBaseModel):
     # Component properties for actions and interactions
     _output_component_property: str = PrivateAttr("children")
 
-    # validator
+    # Re-used validators
     set_actions = _action_validator_factory("cellClicked")
     _validate_callable = validator("figure", allow_reuse=True, always=True)(_process_callable_data_frame)
 
+    @validator("figure")
+    def check_callable_mode(cls, figure):
+        mode = "ag_grid"
+        if mode != figure._mode:
+            raise ValueError(f"CapturedCallable mode mismatch. Expected {mode} but got {figure._mode}.")
+        return figure
```

Comment:
This works under the assumption that the model's `type` is always equal to `figure._mode`. Would it work if users create CustomAgGridTable with `type: Literal["custom_ag_grid"] = "custom_ag_grid"`?

Reply:
Exactly, this would not work.

There is then also the added problem that the two need strings need to be identical, but we may want for some reason decouple the `capture` mode from the model `type`.

So unless there is a way to pass an argument to a re-usable validator, I would go with this approach?

Reply:
There's always the possibility that someone might want to customise the `build` for purpose e.g. to add a table subtitle or to get rid of loading parent div idk.

Reply:
Hm in this case I would err on the side of being more stable at the cost of duplication. It seems quite cheap to have this stability 👍

Reply:
Max, have you tried making a reusable validator with a propagated argument, similar to what we have for the actions validation - `_action_validator_factory`?

Reply:
That does work indeed - maybe the best solution. Although I would argue it does not save that much code :)